### PR TITLE
[Merged by Bors] - doc: add an algebraic number theory section to Mathlib overview

### PR DESCRIPTION
**Pull request by xroblot**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by tb65536** [comment]
Thanks! Looks a bit silly, but there are other autogenerated instance names in this file, and they get turned into links, so it's probably fine.

maintainer merge

File: docs/overview.yaml [hunk 1]
```diff
@@ -139,6 +139,10 @@ General algebra:
   Transcendental numbers:
     Liouville's theorem on existence of transcendental numbers: 'transcendental_liouvilleNumber'
 
+  Algebraic Number Theory:
+    Finiteness of the class number: 'NumberField.RingOfIntegers.instFintypeClassGroupSubtypeMemSubalgebraIntToCommSemiringInstCommRingIntToSemiringToCommSemiringToCommRingToEuclideanDomainAlgebraIntToRingToDivisionRingInstMembershipInstSetLikeSubalgebraRingOfIntegersToCommRingIsDomainToIsDomainIsDedekindDomainIsDomainTo_principal_ideal_domain'
```
Comment (jcommelin) [inline docs/overview.yaml:143]: Hmm, I would much prefer that we rename this celebrated result to something a bit more readable. Would you mind doing that in this same PR?

**Review by jcommelin — COMMENTED** [review]
I think that we shouldn't explicitly reference auto-generated names. I know this has happened before, but just like with auto-generated variable names, it's not the most reliable thing to do.
And in this case the name is also ridiculously long. So please give the instance an explicit name like `NumberField.RingOfIntegers.FintypeClassGroup`. After that, please merge.

bors d+

**Pull request closed without merging**